src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -18,12 +19,17 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
      */
     public DeleteCommand parse(String args) throws ParseException {
         try {
-            Index index = ParserUtil.parseIndex(args);
-            return new DeleteCommand(index);
+            // check if arg is a phone number and initializes Delete Command with args
+            Phone phone = ParserUtil.parsePhone(args);
```

Comment:
I can see a potential error here. I think we need to implement a prefix to differentiate between index and phone number.

I don't think this method will work with higher number of index such as 2024 because parsePhone check if the digit is at least 3 digit long.

I think we just leave it for now, can discuss further on Sunday.

Reply:
The current idea I have in mind is adding prefixes. For example, the command is able to parse all the prefixes and the user have to supply a minimum of 1 prefix and value such as delete i/[index].

When the user write delete n/John a/nus lmao, it will narrow the delete and I think this prefix mechanism can be extended further for batch delete and such.